src/components/numeral-date/numeral-date.test.js

```diff
@@ -0,0 +1,487 @@
+        .should("have.attr", "data-role", CHARACTERS.STANDARD);
+    });
+
+    it("should render NumeralDate with id prop", () => {
+      CypressMountWithProviders(
+        <NumeralDateComponent id={CHARACTERS.STANDARD} />
+      );
+
+      numeralDateInputByPosition(0).should(
+        "have.attr",
+        "id",
+        CHARACTERS.STANDARD
+      );
+    });
+
+    it.each([CHARACTERS.SPECIALCHARACTERS, CHARACTERS.DIACRITICS])(
+      "should render NumeralDate with label prop set to %s",
+      (label) => {
+        CypressMountWithProviders(<NumeralDateComponent label={label} />);
+
+        getDataElementByValue("label").should("have.text", label);
+      }
+    );
+
+    it("should render NumeralDate with labelInline prop", () => {
+      CypressMountWithProviders(<NumeralDateComponent labelInline />);
+
+      getDataElementByValue("label")
+        .parent()
+        .should("have.css", "justify-content", "flex-end");
+    });
+
+    it.each([
+      ["left", "start"],
+      ["right", "end"],
+    ])(
+      "should render NumeralDate with labelAlign prop set to %s",
+      (labelAlign, cssValue) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent labelInline labelAlign={labelAlign} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should(($element) =>
+            expect($element).to.have.css("justify-content", `flex-${cssValue}`)
+          );
+      }
+    );
+
+    it.each([CHARACTERS.SPECIALCHARACTERS, CHARACTERS.DIACRITICS])(
+      "should render NumeralDate with labelHelp prop set to %s",
+      (labelHelp) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent labelInline labelHelp={labelHelp} />
+        );
+
+        getDataElementByValue("question").trigger("mouseover");
+        tooltipPreview().should("have.text", labelHelp);
+      }
+    );
+
+    it.each([
+      [1, "8px"],
+      [2, "16px"],
+    ])(
+      "should render NumeralDate with labelSpacing prop set to %s",
+      (spacing, padding) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent labelInline labelSpacing={spacing} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "padding-right", padding);
+      }
+    );
+
+    it.each([
+      ["10", "90", "135px"],
+      ["30", "70", "405px"],
+      ["80", "20", "1080px"],
+    ])(
+      "should use %s as labelWidth, %s as inputWidth and render it with correct label and input width ratios",
+      (label, input, labelRatio) => {
+        CypressMountWithProviders(
+          <NumeralDateComponent
+            labelInline
+            labelWidth={label}
+            inputWidth={input}
+          />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .should("have.css", "width", labelRatio);
+
+        getDataElementByValue("input")
+          .parent()
+          .should("have.css", "width", "58px");
```

Comment:
Is inputWidth a prop that can be changed?

Reply:
deleted